scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -1613,3 +1613,246 @@ cdef class DiscreteAliasUrn(Method):
             pv_view = _validate_pv(dist)
 
         return pv_view, domain
+
+
+cdef class NaiveRatioUniforms(Method):
+    r"""
+    NaiveRatioUniforms(dist, center=0, *, domain=None, r=1.0, u_min=None, u_max=None, v_max=None, random_state=None)
+
+    Naive Ratio-Of-Uniforms (NROU) Method.
+
+    NROU is an implementation of the (generalized) ratio-of-uniforms method
+    which uses (minimal) bounding rectangles ([3]_, [5]_). It uses a positive
+    control parameter `r` for adjusting the algorithm to the given distribution
+    to improve performance and/or to make this method applicable. Larger values
+    of `r` increase the class of distributions for which the method works at
+    the expense of a higher rejection constant. For computational reasons,
+    ``r=1`` should be used if possible.
+
+    Parameters
+    ----------
+    dist : object
+        An instance of a class with ``pdf`` method.
+
+        * ``pdf``: PDF of the distribution. The signature of the PDF is
+          expected to be: ``def pdf(self, x: float) -> float``. i.e.
+          the PDF should accept a Python float and the parameters and
+          return a Python float. It doesn't need to integrate to 1 i.e.
+          the PDF doesn't need to be normalized.
+
+    center : float, optional
+        Approximate location of the mode or the mean of the distribution.
+        This location provides some information about the main part of the
+        PDF and is used to avoid numerical problems. Default is ``None``.
+
+    domain : list or tuple of length 2, optional
+        The support of the distribution.
+        Default is ``None``. When ``None``:
+
+        * If a ``support`` method is provided by the distribution object
+          `dist`, it is used to set the domain of the distribution.
+        * Otherwise the support is assumed to be :math:`(-\infty, \infty)`.
+
+    r : float, optional
+        Set parameter ``r > 0`` for generalized Ratio-Of-Uniforms method.
+        The default is 1.0.
+
+    u_min: float
+        Sets the left boundary of the bounding rectangle, see details under
+        Notes. If necessary, consider computing the bound numerically using
+        methods in `scipy.optimize`. Important: The bound has to be
+        provided for ``dist.pdf(x - center)``.
```

Comment:
Would it not be possible to allow the user to provide the bound for `x` rather than `x-center`? 
If so, I think we should discuss which would be the more natural interface.

Reply:
`c` in `rvs_ratio_uniform` corresponds to `center`, the generalized version with r is not implemented in the old version

And yes, u and v have switched roles... UNU.RAN does not follow the notation introduced in the original article of Kindernan/Monahan. That is a bit confusing but the examples should help to understand the use of u and v